api/solarperformanceinsight_api/compute.py

```diff
@@ -444,3 +501,182 @@ def compare_expected_and_actual(job: models.StoredJob, si: storage.StorageInterf
+        performance_granularity=(
+            job_params.predicted_data_parameters.performance_granularity
+        ),
+    )
+    ref_pdc_gen = generate_job_performance_data(
+        job,
+        si,
+        types=[models.JobDataTypeEnum.predicted_performance_dc],
+        performance_granularity=(
+            job_params.predicted_data_parameters.performance_granularity
+        ),
+    )
+    results_list = []
+    # Loop through at the inverter level
+    for i, (chain, ref_weather, actual_weather, ref_pac, ref_pdc,) in enumerate(
+        zip_longest(
+            chains,
+            ref_weather_gen,
+            actual_weather_gen,
+            ref_pac_gen,
+            ref_pdc_gen,
+        )
+    ):
+        # use the ModelChain for converting any temperature to cell temperature and
+        # converting irradiance to POA. In the future, could only run these conversions
+        # instead of full modelchain calculation.
+        # since it sets class properties, copy for the actuals calculations
+        chain_actual = deepcopy(chain)
+        pac0 = job.definition.system_definition.inverters[i].inverter_parameters._pac0
+        num_arrays = len(chain.system.arrays)
+        gammas = [
+            arr._gamma for arr in job.definition.system_definition.inverters[i].arrays
+        ]
+        if any([g is None for g in gammas]):
+            raise TypeError(
+                "Currently unable to compare predicted and actual performance for "
+                "PVsyst specified arrays."
+            )
+        if data_available == models.PredictedDataEnum.weather_only:  # 2A-4
+            db_results, _ = process_single_modelchain(
+                chain, ref_weather, ref_model_method, tshift, i
+            )
+            results_list += db_results
+            ref_pdc = adjust(_get_mc_dc(chain.results, num_arrays))  # type: ignore
+        else:
+            # run chain on ref weather
+            getattr(chain, ref_model_method)(
+                [d.shift(freq=tshift) for d in ref_weather]  # type: ignore
+            )
+
+        # run chain on actual weather
+        getattr(chain_actual, actual_model_method)(
+            [d.shift(freq=tshift) for d in actual_weather]  # type: ignore
+        )
+        # use pvlib.modelchain._irrad_for_celltemp that returns POA global if available
+        # otherwise uses effective irradiance
+        poa_ref = _irrad_for_celltemp(
+            chain.results.total_irrad, chain.results.effective_irradiance
+        )
+        poa_actual = _irrad_for_celltemp(
+            chain_actual.results.total_irrad, chain_actual.results.effective_irradiance
+        )
+        # use cell temperature from the pvlib modelchain
+        # If air temp + wind speed or module temperature were supplied, they are
+        # converted to cell temperature via the array's temperature model
+        t_ref = chain.results.cell_temperature
+        t_actual = chain_actual.results.cell_temperature
```

Comment:
test this specifically when providing module_temperature for the two temp models